core/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -91,8 +91,8 @@
     public PluginsService(Settings settings, Path modulesDirectory, Path pluginsDirectory, Collection<Class<? extends Plugin>> classpathPlugins) {
         super(settings);
 
-        List<Tuple<PluginInfo, Plugin>> pluginsLoaded = new ArrayList<>();
-        List<PluginInfo> pluginsList = new ArrayList<>();
+        List<Tuple<PluginInfo, Plugin>> pluginsLoaded = new ArrayList<>(classpathPlugins.size());
+        List<PluginInfo> pluginsList = new ArrayList<>(classpathPlugins.size());
```

Comment:
These aren't right, we load more than just classpath plugins, it's not worth jumping through hoops trying to size them correctly, this code is executed exactly once in the life of an Elasticsearch server or transport client instance. Let's revert this change.